tests/test_db.py

```diff
@@ -34,14 +37,24 @@ def __init__(self, data, status, recorder):
         self._data = data
         self._status = status
         self._recorder = recorder
+        self._etag = '0'
 
     def send(self, request, **kwargs):
+        if 'if-match' in request.headers and request.headers['if-match'] != self._etag:
+            response = Response()
+            response._content = request.body
+            response.headers = CaseInsensitiveDict({'ETag': self._etag})
+            request_exception = RequestException(response=response)
+            raise db.ApiCallError('', request_exception)
```

Comment:
I think it's more accurate to raise the `RequestException` here, and let the `db` module handle it. Or better yet, return a response with status code 412, along with the correct headers and payload.